### ISSUE TITLE
Get current layer name from Kmonad 

### ISSUE THREAD

**Issue by julianinsua** [kmonad/kmonad#769]
Hi! I'm quite new to kmonad and I enjoy it. Thanks for the great job! You saved my wrists! 
That being said, I wanted to ask/suggest something. 
I wanted a little block on my main bar with the name of my current layer on kmonad. I wanted this because I still find myself lost between layers. I thought I could connect to kmonad somehow, through a socket or something, to get the current layer. But I couldn't find any way to do it.
I created a workaround using a command button with a script (maybe that's the way to do it). The script just writes an argument to a file. I made the layer switch button both trigger the script with the name of the layer as an argument and the layer switch/toggle/next. something like :
```lisp
(defalias
   rnum (cmd-button "./scripts/save-current-layer numbers") ;; Run the script to save layer name to file
   num   (around @rnum (layer-switch numbers) )  ;; Trigger both the script and the layer-switch
) 
```

Finally, I get my bar to read the file and display the name. But I don't know if there's a better way to do this. It feels hacky to tell the truth.

Is there a better way to do this? Is there a way to get information from kmonad? 
Thanks in advance!

**Info:**
OS: Linux
Kernel: 6.6.1-arch1-1
Kmonad: 0.4.2, commit 3da3165c413225d924a55933b723ff8ee00f9231

**Comment by ArielHorwitz** [comment]
I had run into this exact same issue. At first I solved it the same way, writing the layer name to a file. I was using my own custom script for `i3bar` which would read this file on a short interval (quick feedback was high priority). This turned out to be quite intensive and taxing, and I could not be bothered to improve a bash script checking the file's modification date etc, just for an indicator on my statusbar.

When I recently switched to polybar, I found out that one of the custom modules -- `ipc` -- was perfect for this task. I still do invoke a bash script, but I have found this completely satisfactory, and still with the quick feedback I desire.

So, I don't think this is any less "hacky" but I gotta say I'm very pleased and have completely stopped thinking about it.

Here is the process from [cmd-button](https://github.com/ArielHorwitz/iukbtw/blob/master/etc/kmd/0base.hs#L10) in KMonad, to [bash script](https://github.com/ArielHorwitz/iukbtw/blob/master/bin/setlayer.sh#L6), to [polybar config](https://github.com/ArielHorwitz/iukbtw/blob/master/home/.config/polybar/config.ini#L179).

Tip: I also change the gamma of the monitor with `xrandr` (changing the color tint) for extra feedback when my keyboard is in "base" mode or "text" mode.

Another tangentially related [discussion](https://github.com/kmonad/kmonad/discussions/711) I opened.
